entrypoint/help_utils_test.go

```diff
@@ -0,0 +1,82 @@
+package entrypoint
+
+import (
+	"regexp"
+
+	"github.com/stretchr/testify/assert"
+	"testing"
+)
+
+var splitKeepDelimiterTests = []struct {
+	in  string
+	out []string
+}{
+	{"one two three", []string{"one ", "two ", "three"}},
+	{"one\ttwo\tthree", []string{"one\t", "two\t", "three"}},
+	{"one\n two    three\t\n ", []string{"one\n ", "two    ", "three\t\n "}},
+	{"onetwothree\n", []string{"onetwothree\n"}},
+	{"\nonetwothree", []string{"\n", "onetwothree"}},
+}
+
+func TestSplitKeepDelimiter(t *testing.T) {
+	for _, tt := range splitKeepDelimiterTests {
+		t.Run(tt.in, func(t *testing.T) {
+			re := regexp.MustCompile("\\s+")
+			assert.Equal(t, SplitKeepDelimiter(re, tt.in), tt.out)
+		})
+	}
+}
+
+var determineTabTests = []struct {
+	in    string
+	delim string
+	out   string
+}{
+	{"   o three", "\t", "   "},
+	{"\to  three", "\t", "\t"},
+	{"o three", "\t", ""},
+	{"  one\ttwo", "\t", "     \t"},
+	{"  \ttwo", "\t", "  \t"},
+	{"  hello|world", "\\|", "        "},
+}
+
+func TestHelpTableAwareDetermineIndent(t *testing.T) {
+	for _, tt := range determineTabTests {
+		t.Run(tt.in, func(t *testing.T) {
+			assert.Equal(t, HelpTableAwareDetermineIndent(tt.in, tt.delim), tt.out)
+		})
+	}
+}
+
+var wrapTextTests = []struct {
```

Comment:
Probably a good idea to add some real-world help text (with many lines) as tests, such as the ones in the PR description.